netmiko/cisco/cisco_vxr_ssh.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import re
+import time
+import logging
+import os
+import paramiko
+
+from netmiko.cisco.cisco_xr import CiscoXrSSH
+from netmiko.netmiko_globals import BACKSPACE_CHAR
+from netmiko.utilities import get_structured_data
+
+work_dir = os.getenv('CAFYKIT_WORK_DIR')
+if work_dir:
+    formatter = logging.Formatter(fmt='%(asctime)s %(levelname)-8s %(message)s',
+                                  datefmt='%Y-%m-%d %H:%M:%S')
+    paramiko_log_folder = os.path.join(work_dir, 'test_paramiko.log')
+    paramiko.util.log_to_file(paramiko_log_folder, level="DEBUG")
+    fh = logging.FileHandler(paramiko_log_folder)
+    fh.setFormatter(formatter)
+    log = logging.getLogger("netmiko")
+    log.addHandler(fh)
+    log.setLevel(logging.DEBUG)
+    log.addHandler(fh)
+else:
+    from netmiko import log
+
+
+class CiscoVxrSSH(CiscoXrSSH):
+    """
+    CiscoVxrSSH is based of CiscoXrSSH -- CiscoBaseConnection
+    """
+
+    def __init__(self,
+                 **kwargs):
+        """Constructor
+        """
+        # 30 minutes
+        self.max_read_timeout = kwargs.get('max_read_timeout', 1800)
+        super().__init__(**kwargs)
+
+    def find_prompt(self, delay_factor=1):
+        """Finds the current network device prompt, last line only.
+
+        :param delay_factor: See __init__: global_delay_factor
+        :type delay_factor: int
+        """
+        delay_factor = self.select_delay_factor(delay_factor)
+        self.clear_buffer()
+        self.write_channel(self.RETURN)
+        time.sleep(delay_factor * 0.1)
+        # Initial attempt to get prompt
+        prompt = self.read_channel()
+        vxr_pattern = "last login"
+        if vxr_pattern in prompt.lower():
+            time.sleep((delay_factor * 0.1) + 3)
+            prompt = self.read_channel()
+        autocommand_pattern = "executing autocommand"
+        if autocommand_pattern in prompt.lower():
+            time.sleep((delay_factor * 0.1) + 5)
+            prompt = self.read_channel()
+        cxr_pattern = "last switch-over"
+        if cxr_pattern in prompt.lower():
+            time.sleep((delay_factor * 0.1) + 3)
+            prompt = self.read_channel()
+        if self.ansi_escape_codes:
+            prompt = self.strip_ansi_escape_codes(prompt)
+
+        # Check if the only thing you received was a newline
+        prompt = prompt.strip()
```

Comment:
no seperate need for this set of commands, this should be part of while loop.
delay_factor : no role .
the while loop is similar while self.is_alive and (current_time - start_time) < self.max_read_timeput and not prompt: